### ISSUE TITLE
Content-Type header's serialization effect might get omitted depending on casing

### ISSUE THREAD

**Issue by tgr** [wikimedia/preq#29]
The README says:
> `body`: `String`, `Buffer` or `Object`. If an object is supplied, the serialization depends on the `content-type` header. 

but in practice it rather depends on the `headers['content-type']` field. Compare
<pre lang="javascript">
preq.post({
    uri: 'https://example.com/',
    headers: {
        'content-type': 'application/json',
    },
    body: {},
});
</pre>
and
<pre lang="javascript">
preq.post({
    uri: 'https://example.com/',
    headers: {
        'Content-Type': 'application/json',
    },
    body: {},
});
</pre>
which I would expect to result in the same request as HTTP generally ignores header case. Actual result is that the first works, in the second no serizalization is applied and `request` throws an error about the body not being a string.

**Comment by d00rman** [comment]
@tgr I can't reproduce this problem. Both of your requests succeed for me.

**Comment by d00rman** [comment]
Actually, ignore my last comment. There is obviously difference in how preq is set up depending on the headers' casing, which is wrong and we should fix that. However, while investigating this, I came across a weird behaviour. I created [a simple script to contact the MW API](https://gist.github.com/d00rman/1d5309cb112494b0b247884f8e552d09), and using `content-type` I get the API's help page, whereas if I set the header name to `Content-Type`, I get the appropriate response, which is rather weird, as this is the case where the body is not stringified. Note that both requests succeed as far as `preq` and `request` are concerned, so in that sense I still cannot reproduce the behaviour described in this ticket.
